### PR TITLE
[netcore] Make `Assembly.Load (new AssemblyName (corlib_name + ".dll"))` work again

### PR DESCRIPTION
**Pull request by lambdageek**
Partly revert a change in https://github.com/mono/mono/pull/12478

CoreCLR treats `Assembly.Load (new AssemblyName ("System.Private.CoreLib.dll"))`
the same as `Assembly.Load (new AssemblyName ("System.Private.CoreLib"))`  (and
.NET Framework behaves likewise with "mscorlib.dll").

### CODE REVIEW

File: mono/metadata/assembly.c [hunk 1]
```diff
@@ -4304,7 +4304,7 @@ mono_assembly_load_full_gac_base_default (MonoAssemblyName *aname,
 	 */
 	gboolean name_is_corlib = strcmp (aname->name, MONO_ASSEMBLY_CORLIB_NAME) == 0;
 #ifndef NETCORE_LOADER
-	name_is_corlib |= strcmp (aname->name, "mscorlib.dll") == 0;
+	name_is_corlib = name_is_corlib || strcmp (aname->name, "mscorlib.dll") == 0;
```
Comment (marek-safar) [inline mono/metadata/assembly.c]: Why is this needed for desktop version only?

**Comment by lambdageek** [comment]
@marek-safar Thanks for asking about this.  Actually, looks like CoreCLR treats `Assembly.Load (new AssemblyName ("System.Private.CoreLib.dll"))` the same as `Assembly.Load (new AssemblyName ("System.Private.CoreLib"))`.  (Also checked that other assembly names, e.g. `"System.Runtime"` don't get this special treatment).

So I need to change this.

**Comment by lambdageek** [comment]
Updated the PR
